src/build_id.cc

```diff
@@ -12,13 +12,11 @@ namespace ddprof {
 BuildIdStr format_build_id(BuildIdSpan build_id_span) {
   std::string build_id_str;
   // We encode every char on 2 hexa string characters
-  build_id_str.resize(build_id_span.size() * 2 + 1);
+  build_id_str.resize(build_id_span.size() * 2);
   for (unsigned i = 0; i < build_id_span.size(); ++i) {
-    snprintf(&build_id_str[static_cast<unsigned>(2 * i)], 3, "%02x",
-             static_cast<unsigned>(build_id_span[i]));
+    (void)snprintf(&build_id_str[static_cast<unsigned>(2 * i)], 3, "%02x",
```

Comment:
I don't understand why this would not do an invalid write
Will (2*build_id_span.size()) with an size of 3 not invalid write ?

Reply:
We talked about this. I ~agree that this is guaranteed to be OK by standard. Though it is still confusing.
I'll let you decide if you want to fix this.

Reply:
Changed the implementation to something simpler.